### PR TITLE
Fix signup page

### PR DESCRIPTION
**Pull request by jreis13**
Why:
* User interviews showed that users prefer having the name and email inputs in the same page

How:
* Refactor signup flow to match the design (moved the name form and email form into a single `DetailsForm`)
* Input boxes now have labels to match the design
* Update the `placeholders` on the inputs
* Remove `STEP_4` as we only have 3 steps now
* Add constraint so that the user has to type an '@' so that we let him move forward past the email input

### CODE REVIEW

File: frontend/src/components/Forms/FormLayout/index.module.css [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .headerWrapper {
-  padding: 0 20px;
+  /* padding: 0 20px; */
```
Comment (joaojusto) [inline frontend/src/components/Forms/FormLayout/index.module.css:2]: If this is not needed please remove it

File: frontend/src/pages/SignupPage/index.js [hunk 2]
```diff
@@ -244,9 +229,10 @@ const SignupPage = () => {
       <div className={styles.buttonWrapper}>
         <Button
           isDisabled={
```
Comment (joaojusto) [inline frontend/src/pages/SignupPage/index.js:231]: Can we extract this into a variable? It's a bit verbose to be inline

**Review by joaojusto — CHANGES_REQUESTED** [review]
Looks good, just minor details

File: frontend/src/components/Forms/FormLayout/index.module.css (hunk 1 above)
Comment (jreis13) [reply frontend/src/components/Forms/FormLayout/index.module.css:2]: You're right, on it.

**Review by joaojusto — APPROVED** [review]
:shipit:

**Pull request merged**